02-standard-library/activityMonitor.ts

```diff
@@ -0,0 +1,41 @@
+import { exec } from "child_process";
+import * as fs from "fs";
+
+const logFile: string = "activityMonitor.log";
+
+function logActivity(): void {
+  let command;
+  if (process.platform === "win32") {
+    command =
+      "powershell \"Get-Process | Sort-Object CPU -Descending | Select-Object -Property Name, CPU, WorkingSet -First 1 | ForEach-Object { $_.Name + ' ' + $_.CPU + ' ' + $_.WorkingSet }\"";
+  } else if (process.platform === "linux" || process.platform === "darwin") {
```

Comment:
If the platform is not win32, linux nor darwin - which are only a few of the options - command will stay undefined and cause an error. For this, your code did not run on my machine. Please change this logic, so that the command will not be undefined under any circumstance. As mentioned, all platforms are UNIX based, so it's enough to filter out win32 and apply the same command for the rest.